### ISSUE TITLE
[Bug] Wrong Authority created in CreateAuthorityForRequest method call

### ISSUE THREAD

**Issue by wziska** [AzureAD/microsoft-authentication-library-for-dotnet#1606]
**Which Version of MSAL are you using ?**
MSAL 4.7.1

**Platform**
net45, xamarin android, xamarin iOS

**What authentication flow has the issue?**
Any

**Is this a new or existing app?**
The app is in production, I haven't upgraded MSAL, but started seeing this issue

**Repro**

```csharp
            var publicClientBuilder = PublicClientApplicationBuilder.Create(ClientId)
                .WithLogging((level, msg, pii) => { Console.WriteLine(msg); });

            var clientApp = publicClientBuilder.Build();

            var getTokenResult = await clientApp.AcquireTokenInteractive(Scopes)
                 .WithPrompt(Prompt.SelectAccount)
                 .WithB2CAuthority(CreateAuthorityString("https://login.microsoftonline.com/tfp/<tentant>.onmicrosoft.com/<policy>"))
                 .ExecuteAsync();

```

**Expected behavior**
As I added WithB2CAuthority I expect that in Authority.CreateAuthorityForRequest method new B2CAuthority class is instantiated.

**Actual behavior**
AadAuthority instance is created in Authority.CreateAuthorityForRequest call.

**Possible Solution**
It seems this behavior was introduced in https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/1456
Accordingly to comment in https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/faabb9c5a42a904406db2a7c3bed529438d60fe1/src/client/Microsoft.Identity.Client/Instance/Authority.cs#L37 we should check AuthorityType in requestAuthorityInfo (if exists). Now requestAuthorityInfo is ignored and only [configAuthorityInfo.AuthorityType is considered](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/faabb9c5a42a904406db2a7c3bed529438d60fe1/src/client/Microsoft.Identity.Client/Instance/Authority.cs#L52) .

**Comment by bgavrilMS** [comment]
Ack, I can see the problem. To work around it you need to specify an authority when you create the `PublicClientApplication`. It doesn't matter the policy, as the policy specified in `AcquireTokenInteractive` will be taken into account.

**Comment by trwalke** [comment]
Fixed in 4.9.0
https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/releases/tag/4.9.0
